src/unix/core.c

```diff
@@ -53,6 +53,9 @@
 # include <mach-o/dyld.h> /* _NSGetExecutablePath */
 # include <sys/filio.h>
 # include <sys/ioctl.h>
+# if defined(MAC_OS_X_VERSION_10_7)
```

Comment:
I don't think we support 10.6 anymore, so we should probably do this unconditionally. @bnoordhuis ?